### PR TITLE
objectify ccxt fix backtesting and some tests

### PR DESCRIPTION
**Pull request by xmatthias**
## Summary
this will fix a few tests in objectify.
Also, rolling back the change of the OHLVC format - this would complicate things a lot and make all saved dataframes worthless.

## Quick changelog

- changed all tests in root and rpc folders
- backtesting / dry mode on binance working
- revert OHLVC format so existing dataframes are not broken

CCXT work is not complete with this, but i hope it helps bring ccxt objectify forward.

Currently mainly interresting for @shusso 

Tests are expected to fail as not all tests are fixed yet.

### CODE REVIEW

File: freqtrade/analyze.py [hunk 1]
```diff
@@ -44,13 +44,13 @@ def parse_ticker_dataframe(ticker: list) -> DataFrame:
         :param ticker: See exchange.get_ticker_history
         :return: DataFrame
         """
-        cols = ['date', 'open', 'high', 'low', 'close', 'volume']
```
Comment (shusso) [inline freqtrade/analyze.py]: I'm perfectly fine having backtesting break for now. The way forwad is the `ccxt` way. We can think of creating a converter that goes through the data and transforms the old data to the new format, or then for backtesting add an temporary intermediate that overwrites the `parse_ticker_dataframe` to use the old format

File: freqtrade/exchange/__init__.py [hunk 2]
```diff
@@ -183,11 +184,34 @@ def get_ticker(pair: str, refresh: Optional[bool] = True) -> dict:
 def get_ticker_history(pair: str, tick_interval) -> List[Dict]:
     # TODO: tickers need to be in format 1m,5m
     # fetch_ohlcv returns an [[datetime,o,h,l,c,v]]
-    if not _API.markets:
-        _API.load_markets()
-    ohlcv = _API.fetch_ohlcv(pair, str(tick_interval)+'m')
+    if 'fetchOHLCV' not in _API.has or not _API.has['fetchOHLCV']:
+        raise OperationalException(
```
Comment (shusso) [inline freqtrade/exchange/__init__.py]: should the `get_ticker_history` raise anything? I think it either gets the ticker history or then returns empty result

File: freqtrade/exchange/__init__.py [hunk 3]
```diff
@@ -183,11 +184,34 @@ def get_ticker(pair: str, refresh: Optional[bool] = True) -> dict:
...
+                'L': candlestick[3],
+                'C': candlestick[4],
+                'V': candlestick[5],
+            })
+        return history_json
+    except IndexError as e:
+        logger.warning('Empty ticker history. Msg %s', str(e))
+        return []
+    except ccxt.NetworkError as e:
+        raise NetworkException(
```
Comment (shusso) [inline freqtrade/exchange/__init__.py]: same as above

File: freqtrade/exchange/__init__.py [hunk 4]
```diff
@@ -183,11 +184,34 @@ def get_ticker(pair: str, refresh: Optional[bool] = True) -> dict:
...
+            'Could not load ticker history due to networking error. Message: {}'.format(e)
+        )
+    except ccxt.BaseError as e:
+        raise OperationalException('Could not fetch ticker data. Msg: {}'.format(e))
```
Comment (shusso) [inline freqtrade/exchange/__init__.py]: same as above

File: freqtrade/tests/test_persistence.py [hunk 5]
```diff
@@ -122,7 +122,7 @@ def test_update_with_bittrex(limit_buy_order, limit_sell_order):
         pair='BTC_ETH',
         stake_amount=0.001,
         fee=0.0025,
-        exchange=Exchanges.BITTREX,
+        exchange='binance',
```
Comment (shusso) [inline freqtrade/tests/test_persistence.py]: Not in scope of this PR but generally in this whole ´ccxt´ work that how do we organize our tests and which one is the "default" exchange..

**Comment by shusso** [comment]
Just a general comment that this is in my opinion of the priority:
- make dry_run work
- make tests pass
- make traiding work
- make tests pass
- make backtesting work
- make hyperopt work

**Comment by vertti** [comment]
Old backtesting data will need to go. So if we really really want to support it, it will be with a separate conversion script. But the new bot will only support using the format that CCXT returns. We definitely do not want to support multiple formats and/or convert between multiple temporary formats due to legacy reasons.

**Comment by xmatthias** [comment]
Good feedback, will get these changes fixed.

@shusso can you put your comment as a issue (or add it to #230 ?)- so we can track next steps, know the direction, and may be able to split work to get the whole thing done faster.

i am a fan of having a few smaller PR's - so i hope this one can be merged soon - having your "task list" disappear into closed Pr's)

File: freqtrade/tests/test_persistence.py (hunk 5 above)
Comment (xmatthias) [reply freqtrade/tests/test_persistence.py]: Good point - i'll revert this to bittrex for now (so bittrex is the default testing exchange throut the tests).

Would it make sense to have a variable in conftest defining this? so we can farily simply support testing over multiple exchanges?

File: freqtrade/exchange/__init__.py (hunk 4 above)
Comment (xmatthias) [reply freqtrade/exchange/__init__.py]: Yes - fixed both this and the above

**Comment by xmatthias** [comment]
Updated the PR with the requested changes.

Backtesting also seems to work, but requires a refresh of the data as no conversation-script exists yet.
Tests still expected to fail - as only some have been taken care off.

**Comment by shusso** [comment]
>  can you put your comment as a issue 
updated #230 with a task list and priority

**Pull request merged**